### PR TITLE
✨ add show truncated content on hover

### PR DESCRIPTION
**Pull request by mholta**
Should be discussed. Should this be a non configurable default?

## What does this pull request change?

- Add meta prop to show truncated content on hover
- Improve TypographyCustom by inheriting cell color from cell. This makes the component agnostic 
of color.

## Why is this pull request needed?

- To improve table when using fixed width

## Issues related to this change:
[AB#328226](https://dev.azure.com/Equinor/5f972ff3-ed9a-4405-9db1-84542b5007a8/_workitems/edit/328226)

### CODE REVIEW

**Comment by cbuv** [comment]
> Should be discussed. Should this be a non configurable default?

I think an opinionated default column is fine if it can be overridden (prop to useDataTable), for now, we should probably make it configurable.

**Review by Simrayz — APPROVED** [review]
LGTM. Leaving it as a configurable option for now is fine. I agree that it maybe should be the default - it depends how content other than text is rendered? 🤔

**Comment by mholta** [comment]
Truncate or wrap

**Review by Simrayz — APPROVED** [review]
LGTM 🚀

File: packages/apollo-components/src/DataTable/useDataTable.tsx [hunk 1]
```diff
@@ -54,9 +45,17 @@ export function useDataTable<T>(props: UseTableData<T>): Table<T> {
       columnVisibility,
     },
     defaultColumn: {
-      cell: (cell) => <TypographyCustom noWrap>{cell.getValue() as any}</TypographyCustom>,
+      cell: ({ cell }) => {
+        let truncateMode = cellConfig?.truncateMode ?? 'hover'
+        if (typeof truncateMode === 'function') truncateMode = truncateMode(cell)
```
Comment (Simrayz) [inline packages/apollo-components/src/DataTable/useDataTable.tsx:50]: Hmm, I would have written it like this:
```suggestion
      cell: ({ cell }) => {
        if (!cellConfig?.truncateMode) return 'hover'
        const truncateMode = typeof cell.truncateMode === 'function' ? truncateMode(cell) : cell.truncateMode
```

**Review by Simrayz — CHANGES_REQUESTED** [review]
A lot of good work here, just one suggested change for clarity 😄

File: packages/apollo-components/src/DataTable/useDataTable.tsx (hunk 1 above)
Comment (mholta) [reply packages/apollo-components/src/DataTable/useDataTable.tsx:50]: Yes 👀

**Review by Simrayz — APPROVED** [review]
Let's go 🚀 🌠

**Pull request merged**